src/Neo/ProtocolSettings.cs

```diff
@@ -131,7 +131,7 @@ public record ProtocolSettings
         /// <returns>The loaded <see cref="ProtocolSettings"/>.</returns>
         public static ProtocolSettings Load(Stream stream)
         {
-            var config = new ConfigurationBuilder().AddJsonStream(stream).Build();
+            using var config = new ConfigurationBuilder().AddJsonStream(stream).Build();
```

Comment:
`ConfigurationBuilder` on build uses `ConfigurationRoot` may have to cast to it

https://github.com/dotnet/runtime/blob/9d5a6a9aa463d6d10b0b0ba6d5982cc82f363dc3/src/libraries/Microsoft.Extensions.Configuration/src/ConfigurationBuilder.cs#L53C24-L53C41

Reply:
It was not a memory leak...

Reply:
How isn't it a memory leak, if the memory for the stream is the still allocated?
https://github.com/dotnet/runtime/blob/9d5a6a9aa463d6d10b0b0ba6d5982cc82f363dc3/src/libraries/Microsoft.Extensions.Configuration/src/ConfigurationRoot.cs#L99C9-L112C10